tests/test_georaster.py

```diff
@@ -557,24 +542,30 @@ def test_copy(self, example: str) -> None:
             assert r.__getattribute__(attr) == r2.__getattribute__(attr)
 
         # Check data array
-        assert geoutils.misc.array_equal(r.data, r2.data, equal_nan=True)
+        assert np.array_equal(r.data, r2.data, equal_nan=True)
 
         # Check dataset_mask array
-        assert np.all(r.data.mask == r2.data.mask)
+        assert np.array_equal(r.data.mask, r2.data.mask)
 
         # -- Third test: if r.data is modified, it does not affect r2.data --
         r.data += 5
-        assert not geoutils.misc.array_equal(r.data, r2.data, equal_nan=True)
+        assert not np.array_equal(r.data.data, r2.data.data, equal_nan=True)
 
         # -- Fourth test: check the new array parameter works with either ndarray filled with NaNs, or masked arrays --
 
         # First, we pass the new array as the masked array, mask and data of the new Raster object should be identical
         r2 = r.copy(new_array=r.data)
         assert r == r2
 
-        # Same when passing the new array as a NaN ndarray
+        # When passing the new array as a NaN ndarray, only the valid data is equal, because masked data is NaN in one
+        # case, and -9999 in the other
         r_arr = gu.spatial_tools.get_array_and_mask(r)[0]
         r2 = r.copy(new_array=r_arr)
+        assert np.ma.allequal(r.data, r2.data)
+        # If a nodata value exists, and we update the NaN pixels to be that nodata value, then the two Rasters should
```

Comment:
This test would not be needed if comparing only valid values and mask in __eq__

Reply:
But then the others tests would have to change. We need to perform both functionalities in any case, see answer to point above.